src/TabBar.tsx

```diff
@@ -291,9 +351,22 @@ export default class TabBar<T extends Route> extends React.Component<
               this.scrollView = el && el.getNode();
             }}
           >
-            {routes.map((route: T) => (
+            {routes.map((route: T, i: number) => (
               <TabBarItem
+                onLayout={({ nativeEvent: { layout } }) => {
+                  if (!dynamicWidth) return;
+
+                  this._actualTabWidths[i] = layout.width || tabWidth;
+
+                  // check if this is the last onLayout call
+                  if (!allDone(this._actualTabWidths, routes.length)) return;
```

Comment:
I'm curious on how this works if one of the tab's width changes/number of tabs change